src/engraving/dom/instrumentname.h

```diff
@@ -64,6 +64,8 @@ class InstrumentName final : public TextBase
     SysStaff* sysStaff() const { return _sysStaff; }
     void setSysStaff(SysStaff* s) { _sysStaff = s; }
 
+    double spatium() const;
```

Comment:
This implicitly overrides EngravingItem::spatium(). You should make EngravingItem::spatium() virtual and override it explicitly. This type of solution may not be perfect either, but at least it is more obvious, and you no longer need that cast inside TextFragment::font

Reply:
Making `EngravingItem::spatium()` virtual was my first approach, but after Michele asked me to do a performance test we found that layout took noticeably longer (3-4%).  I can change the name of the method to avoid confusion, though we'd keep the cast.

Reply:
Yeah, making spatium() virtual is very elegant codewise, but spatium() is an extremely hot function, and my concern was that resolving the vtable at every call would carry a noticeable runtime cost (which seems to be indeed the case), so I advised against making it virtual. I agree that overriding the method implicitely may cause confusion though, so it's probably better to just call it a different name

Reply:
Yes, you are right. You just should rename InstrumentName::spatium()